### PR TITLE
Add documentation

### LINKED ISSUE #18: Add docs
Add doctrings and minimal documentation in `README.md`

### PR DESCRIPTION
**Pull request by pabloferz**
Closes #18.

TODO:
 - [x] Improve `README.md`

### CODE REVIEW

File: src/DLPack.jl [hunk 1]
```diff
@@ -148,6 +192,19 @@ function wrap(manager::DLManagedTensor, foreign)
...
+    wrap(::Type{A}, ::Type{M}, manager::DLManagedTensor, foreign) where {
+        T, N, A <: AbstractArray{T, N}, M <: MemoryLayout
+    }
+
+Takes a `DLManagedTensor` and a corresponding `foreign` tensor, and tries to
+return a zero-copy `array::AbstractArray{T, N}` pointing to the same data in
+`foreign` in a type-inferable way.
+If there is a mismatch between any of the types `T`, `N`, or memory layout `M`
+and the matching properties of `foreign`, an error is thrown.
+For tensors with row-mayor ordering the resulting array will have all
```
Comment (rejuvyesh) [inline src/DLPack.jl:205]:
```suggestion
For tensors with row-major ordering the resulting array will have all
```

File: src/pycall.jl [hunk 2]
```diff
@@ -41,18 +49,46 @@ function DLManagedTensor(po::PyObject)
...
+The resulting tensor will have all dimensions reversed with respect
+to the Julia array.
+"""
 share(A::StridedArray, from_dlpack::PyObject) = share(A, PyObject, from_dlpack)
-#
+
+"""
+    share(A::StridedArray, ::Type{PyObject}, from_dlpack)
+
+Similar to `share(A, from_dlpack::PyObject)`. Use when there is a needed to
```
Comment (rejuvyesh) [inline src/pycall.jl:89]:
```suggestion
Similar to `share(A, from_dlpack::PyObject)`. Use when there is a need to
```

File: src/pythoncall.jl [hunk 3]
```diff
@@ -61,6 +69,12 @@ let
         return tensor
     end
 
+    """
+        share(A::StridedArray, ::Type{Py}, from_dlpack)
+
+    Similar to `share(A, from_dlpack::Py)`. Use when there is a needed to
```
Comment (rejuvyesh) [inline src/pythoncall.jl:75]:
```suggestion
    Similar to `share(A, from_dlpack::Py)`. Use when there is a need to
```

File: src/pythoncall.jl [hunk 4]
```diff
@@ -83,14 +97,36 @@ let
 
 end
 
+"""
+    wrap(o::Py, to_dlpack)
+
+Takes a tensor `o::Py` and a `to_dlpack` function that generates a
+`DLManagedTensor` bundled in a PyCapsule, and returns a zero-copy
+`array::AbstractArray` pointing to the same data in `o`.
+For tensors with row-mayor ordering the resulting array will have all
```
Comment (rejuvyesh) [inline src/pythoncall.jl:106]:
```suggestion
For tensors with row-major ordering the resulting array will have all
```

**Comment by pabloferz** [comment]
@JuliaRegistrator register

**Pull request merged**